### DISCUSSION TITLE
How to use multiple custom click group classes?

### DISCUSSION THREAD

**Discussion by ssbarnea** [fastapi/typer#1579] [Questions]
### First check

I wanted to use two click plugins in parallel but adding my `CustomTyper(typer.Typer)` class allows me to specify only one class to use.

### Description

```
from click_help_colors import HelpColorsCommand, HelpColorsGroup
from click_didyoumean import DYMGroup


class CustomTyper(typer.Typer):
    def __init__(
        self,
        *args,
        cls=DYMGroup,  # <-- i can specify just a single class here
        context_settings={"help_option_names": ["-h", "--help"]},
        **kwargs
    ) -> None:
        super().__init__(*args, cls=cls, context_settings=context_settings, **kwargs)
```

Is it possible to [...]?

### Additional context

I also tried to add another CustomGroup class that inherits from both but it did not had the desired effect as some methods shadows the ones from the other class.

Are such click plugins mutually exclusive or is a way to make use of them in parallel?

**Comment by cihati** [comment]
@ssbarnea Not for typer specifically, but I did this for Click and it worked:

```
class AllGroups(HelpColorsGroup, DYMGroup):
    pass

@click.group(cls=AllGroups, help_headers_color='yellow', help_options_color='green')
def cli() -> None:
    """foo bar"""
```
